### PR TITLE
Adds more documentations around ignoreSemantics deprecations.

### PR DESCRIPTION
**Pull request by chunhtai**
migration guide update https://github.com/flutter/website/pull/9124

fixes the concerns in https://github.com/flutter/flutter/pull/120619
## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -6800,6 +6800,18 @@ class RepaintBoundary extends SingleChildRenderObjectWidget {
 /// semantics subtree is collected. If [ignoring] is true, pointer-related
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart:6800]: would it make sense to move the `{@template flutter.widgets.IgnorePointer.Semantics}` line to above the `## Semantics` header now that it's so big? e.g. consider how it looks in [SliverIgnorePointer](https://master-api.flutter.dev/flutter/widgets/SliverIgnorePointer-class.html).

**Comment by Hixie** [comment]
How will that render for the various places that use the macro, compared to the place that declares it?

(To put it another way, I recommend auditing all the places in the codebase that mention `flutter.widgets.IgnorePointer.Semantics`. I see eleven. From a quick inspection, I expect this PR will result in some of those places not being as clear as we would like. In general when editing a macro one should always verify how it affects all the places that use the macro.)

**Comment by chunhtai** [comment]
@Hixie i rewrote some of the doc, PTAL

**Review by Hixie — APPROVED** [review]
LGTM

Thanks!

**Pull request merged**